util/testutil/dockerd/client/client.go

```diff
@@ -0,0 +1,191 @@
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/docker/docker/api"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/versions"
+	"github.com/docker/go-connections/sockets"
+	"github.com/pkg/errors"
+)
+
+const defaultDockerHost = "unix:///var/run/docker.sock"
+const defaultProto = "unix"
+const defaultAddr = "/var/run/docker.sock"
+
+var ErrRedirect = errors.New("unexpected redirect in response")
+
+type Client struct {
+	scheme            string
```

Comment:
I think almost none of this is actually used. We don't need protos/tls or header/version negotiation. Only thing we need is to dial the UNIX socket and do the upgrade.

We can do this part later if you want. Eg. create a tracking issue for it or we can remove this commit from the current PR for now.

Reply:
Sure fine to move it to another PR

Reply:
Removed for now, will do this in another PR.